src/TagLib/Ogg/XiphComment.cs

```diff
@@ -211,6 +231,13 @@ public void SetField (string key, params string [] values)
 				field_list [key] = result.ToArray ();
 			else
 				field_list.Add (key, result.ToArray ());
+
+			// If the user is overwriting a picture field, reset
+			// the state in the pictures array
+			if (IsPictureField (key)) {
```

Comment:
I'm guessing that the picture field contains 0 or more pictures?

Reply:
The semantics of the `pictures` and `picture_fields_dirty` field are:
* If `pictures` is null, the state of the pictures for this tag is stored in `field_list` for the different `PICTURE_FIELDS`
* If `pictures` is not null:
  * If `picture_fields_dirty` is true: the `pictures` array has changed, and `field_list` is not up-to-date with what's stored in `pictures`.
  * If `picture_fields_dirty` is false: the `pictures` fields has just been loaded from `field_list`, so they both represent the same state.

So the `pictures` array can indeed contain 0 or more pictures, if the user set the `Pictures` property to a 0-length array. There will be an error if the user sets `Pictures` to null though. I'll add a test.

Reply:
Fixed in 2819c6675529c9bba83c1759c46ec5dd91f287cb (this actually triggered a re-parse of the pictures field)